rotkehlchen/chain/polygon/manager.py

```diff
@@ -0,0 +1,204 @@
+            tx_receipt['gasUsed'] = tx_receipt['cumulativeGasUsed']
+            successful = tx_receipt.pop('successful', None)
+            tx_receipt['status'] = 1 if successful else 0
+            tx_receipt['transactionIndex'] = 0
+            txhash = tx_receipt.pop('tx_hash')
+            tx_receipt['hash'] = txhash
+
+            # TODO input and nonce is decoded in Covalent api, encoded in future
+            tx_receipt['input'] = '0x'
+            tx_receipt['nonce'] = 0
+            for index, receipt_log in enumerate(tx_receipt['log_events']):
+                receipt_log['blockNumber'] = block_number
+                receipt_log['logIndex'] = receipt_log.pop('log_offset', None)
+                receipt_log['transactionIndex'] = 0
+                tx_receipt['log_events'][index] = receipt_log
+        except (DeserializationError, ValueError) as e:
+            raise RemoteError(
+                f'Couldnt deserialize transaction receipt '
+                f'data from covalent {tx_receipt} due to {str(e)}',
+            ) from e
+        return tx_receipt
+
+    def call_contract(
+            self,
+            contract_address: ChecksumEthAddress,
+            abi: List,
+            method_name: str,
+            arguments: Optional[List[Any]] = None,
+    ) -> Any:
+        """Performs an eth_call to an ethereum contract
+
+        May raise:
+        - RemoteError if Covalent is used and there is a problem with
+        reaching it or with the returned result
+        - BlockchainQueryError if web3 is used and there is a VM execution error
+        """
+
+        contract = self.w3.eth.contract(address=contract_address, abi=abi)
+        try:
+            method = getattr(contract.caller, method_name)
+            result = method(*arguments if arguments else [])
+        except (ValueError, BadFunctionCallOutput) as e:
+            raise BlockchainQueryError(
+                f'Error doing call on contract {contract_address}: {str(e)}',
+            ) from e
+        return result
+
+    @staticmethod
+    def get_basic_contract_info(address: ChecksumEthAddress) -> Dict[str, Any]:
+        """
+        Query a contract address in pangolin graph node and return basic information as:
+        - Decimals
+        - name
+        - symbol
+        if it is provided in the contract. This method may raise:
+        - BadFunctionCallOutput: If there is an error calling a bad address
+        """
+        properties = ('decimals', 'symbol', 'name')
+        info: Dict[str, Any] = {}
+        try:
+            # Output contains call status and result
+            graph = Graph('https://api.thegraph.com/subgraphs/name/dasconnor/pangolin-dex')
+            output = graph.query(
+                f'''{{token(id:"{address.lower()}"){{
+                    symbol
+                    name
+                    decimals
+                    }}
+                }}
+                ''',
+            )
```

Comment:
We are tying to avoid the usage of subgraphs in the app. This should be changed to query a node for this information as we do in ethereum https://github.com/rotki/rotki/blob/ae10d6b55f6fb6f75e6be8d231f22a422ef19b7f/rotkehlchen/chain/ethereum/manager.py#L1140

Reply:
Done! thanks